### PR TITLE
Create Convolutional_Neural_Network.md

### LINKED ISSUE #1114: Convolutional Neural Networks (CNN)
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md).

### Description

CNNs are a type of neural network specifically designed for processing image and video data. They use convolutional layers to extract features from the input data. CNNs have achieved state-of-the-art results in tasks like image classification, object detection, and image segmentation.

### Self-service

- [X] I'd be willing to address this documentation request myself.

### PR DESCRIPTION
**Pull request by inkerton**
### Description
Convolutional Neural Networks (CNNs) are a class of deep neural networks specifically designed to process data with a grid-like structure, such as images. CNNs are highly effective in tasks like image classification, object detection, and recognition due to their ability to capture spatial hierarchies in data through convolutional layers.

Fixes #1114 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/machine-learning/Convolutional_Neural_Network.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+6. **Output**:  
+   For classification, the output is passed through a softmax layer, which converts the output into class probabilities.
+
+### Split Criteria:
+CNNs split data by progressively extracting features at different layers, with lower layers detecting simple patterns (e.g., edges) and higher layers capturing complex features (e.g., objects).
+
+### Time Complexity:
+- **Training Complexity**:  
+  The time complexity depends on the number of filters, kernel size, and input dimensions. For an image of size \( n \times n \) with \( k \) filters of size \( f \times f \), the time complexity of a convolutional layer is \( O(n^2 \cdot f^2 \cdot k) \).
```
Comment (ajay-dhangar) [inline docs/machine-learning/Convolutional_Neural_Network.md]: replace `\( n \times n \)` with `$ n \times n $`, `\( k \)` with `$k$`, `\( f \times f \)` with `$ f \times f $, and `\( O(n^2 \cdot f^2 \cdot k) \)` with `$ O(n^2 \cdot f^2 \cdot k)$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by inkerton** [comment]
done

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**